ci/do_ci.sh

```diff
@@ -520,6 +520,9 @@ elif [[ "$CI_TARGET" == "verify_distro" ]]; then
     fi
     bazel run "${BAZEL_BUILD_OPTIONS[@]}" //distribution:verify_packages "$PACKAGE_BUILD"
     exit 0
+elif [[ "$CI_TARGET" == "publish" ]]; then
+    GITHUB_TOKEN=TOKEN  bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/project:publish
```

Comment:
@lizan a token is required here

Reply:
It is stored as `GitHubPublicRepoOnlyAccessToken` in azp.

Can you test this PR by pushing this to a non-fork branch and manually triggering a CI workflow in envoy-postsubmit?

Reply:
ive created the branch (`auto-tag--testing`) - actually testing is not so trivial as it is checking the branch etc before running the code - also, if it succeeds it would create a release/tag

Reply:
k - ive added a step at the beginning with  a curl test to see if it can create the release (we will need to delete it after)

Reply:
doesnt appear to work - response is:

```console
{
  "message": "Not Found",
  "documentation_url": "[https://docs.github.com/rest/reference/repos#create-a-release"](https://docs.github.com/rest/reference/repos#create-a-release%22)
}
```

which i believe is the response you get if you dont have creds

https://dev.azure.com/cncf/envoy/_build/results?buildId=114516&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=2c013f53-8531-54ae-b0be-2ca05b3100aa&l=21

Reply:
test code is here https://github.com/envoyproxy/envoy/blob/ce15d1995c24d02f2b604e5c30f94541138fb135/.azure-pipelines/pipelines.yml#L33-L41

Reply:
testing branch is here https://github.com/envoyproxy/envoy/tree/auto-tag--testing